api/v1alpha1/randomsecret_types.go

```diff
@@ -306,7 +306,7 @@ func (r *RandomSecret) validateEitherPasswordPolicyReferenceOrInline() error {
 func (r *RandomSecret) validateInlinePasswordPolicyFormat() error {
 	if r.Spec.SecretFormat.InlinePasswordPolicy != "" {
 		passwordPolicyFormat := &PasswordPolicyFormat{}
-		return hclsimple.Decode(r.Spec.SecretKey, []byte(r.Spec.SecretFormat.InlinePasswordPolicy), nil, passwordPolicyFormat)
```

Comment:
I'd recommend ensuring that a valid extension is present on the value retrieved from `secretKey` property. If it is not present, it can be added to sure that the requirement enforced by this method is satisfied instead of a hardcoded value

Reply:
done